plone/app/linkintegrity/handlers.py

```diff
@@ -111,6 +114,25 @@ def modifiedContent(obj, event):
         updateReferences(obj, refs)
 
 
+def removedContent(obj, event):
+    if not check_linkintegrity_dependencies(obj):
+        return
+
+    int_id = ensure_intid(obj)
```

Comment:
We don't really want to add a new intid to an object that's about to be removed. Should this just do the part of ensure_intid that checks for an existing intid and not the part that adds a new one if needed?

Reply:
Yes definitly. You are right.